homeassistant/components/asuswrt/config_flow.py

```diff
@@ -101,71 +127,98 @@ def _get_ip(host: str) -> str | None:
 
 
 class AsusWrtFlowHandler(ConfigFlow, domain=DOMAIN):
-    """Handle a config flow."""
+    """Handle a config flow for AsusWRT."""
 
     VERSION = 1
 
+    def __init__(self) -> None:
+        """Initialize the AsusWrt config flow."""
+        self._config_data: dict[str, Any] = {}
+        self._unique_id: str | None = None
+
     @callback
-    def _show_setup_form(
-        self,
-        user_input: dict[str, Any] | None = None,
-        errors: dict[str, str] | None = None,
-    ) -> FlowResult:
+    def _show_setup_form(self, error: str | None = None) -> FlowResult:
         """Show the setup form to the user."""
 
-        if user_input is None:
-            user_input = {}
+        user_input = self._config_data
 
-        adv_schema = {}
-        conf_password = vol.Required(CONF_PASSWORD)
         if self.show_advanced_options:
-            conf_password = vol.Optional(CONF_PASSWORD)
-            adv_schema[vol.Optional(CONF_PORT)] = cv.port
-            adv_schema[vol.Optional(CONF_SSH_KEY)] = str
+            add_schema = {
+                vol.Optional(CONF_PASSWORD): str,
+                vol.Optional(CONF_PORT): cv.port,
+                vol.Optional(CONF_SSH_KEY): str,
```

Comment:
Just one comment before going ahead with this: in this way we will have a more generic error:

```
 User input malformed: two or more values in the same group of exclusion 'pass_key' @ data[<pass_key>]
```

instead of a clear explanation.
Do we prefer in any case this?

Reply:
Maybe exist a way to provide a more descriptive message (https://alecthomas.github.io/voluptuous/docs/_build/html/voluptuous.html#voluptuous.schema_builder.Exclusive) ... i'll check

Reply:
Replaced with exclusive